src/Type/Php/PregSplitDynamicReturnTypeExtension.php

```diff
@@ -36,17 +49,161 @@ public function isFunctionSupported(FunctionReflection $functionReflection): boo
 
 	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
 	{
-		$flagsArg = $functionCall->getArgs()[3] ?? null;
+		$args = $functionCall->getArgs();
+		if (count($args) < 2) {
+			return null;
+		}
+		$patternArg = $args[0];
+		$subjectArg = $args[1];
+		$limitArg = $args[2] ?? null;
+		$capturesOffset = $args[3] ?? null;
+		$patternType = $scope->getType($patternArg->value);
+		$patternConstantTypes = $patternType->getConstantStrings();
+		if (count($patternConstantTypes) > 0) {
+			foreach ($patternConstantTypes as $patternConstantType) {
+				if ($this->isValidPattern($patternConstantType->getValue()) === false) {
+					return new ErrorType();
+				}
+			}
+		}
+
+		$subjectType = $scope->getType($subjectArg->value);
+		$subjectConstantTypes = $subjectType->getConstantStrings();
+
+		$limits = [];
+		if ($limitArg === null) {
```

Comment:
similar to https://github.com/phpstan/phpstan-src/pull/3970/files we should not only handle a absent argument, but also a explicit `null` passed:

e.g. `preg_split($pattern, $subject, null, PREG_SPLIT_NO_EMPTY);`